<launch>

  <!-- Launch the ekf node -->
  <node pkg="snowmower_localization" type="ekf_node" name="ekf" output="screen">
    <param name="base_frame" value="base_link" type="str"/>
    <param name="map_frame" value="map" type="str"/>
    <!-- All matricies are created from arrays in column major order. -->

    <!-- State and Covariance Initialization -->
    <rosparam param="state">[2.7432, 2.7432 , 1.570796327, 0.0 ,0.0 ,0.0]</rosparam>
    <rosparam param="covariance">[100, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 100]</rosparam>

    <!-- System Initialization -->
    <rosparam param="Q">[0.01, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0, 0, 0.01]</rosparam>

    <!-- DecaWave Initialization -->
    <!-- Entries in the R matrix for the decawave beacons -->
    <rosparam param="DW_R">[0.001, 0, 0, 0, 0, 0.001, 0, 0, 0, 0, 0.001, 0, 0, 0, 0, 0.001]</rosparam>
    <!-- Beacon Location x-y [x1, y1, x2, y2, x3, y3, x4, y4] -->
    <rosparam param="DW_Beacon_Loc">[0.0, 0.0, 0.0, 5.4864, 5.4864, 5.4864, 5.4864, 0.0]</rosparam>
    <param name="DW_offset_x" value="-0.4" type="double"/>
    <param name="DW_offset_y" value="-0.1" type="double"/>

    <!-- Wheel Encoder Initialization -->
    <!-- Entries in the R matrix for the wheel encoders -->
    <rosparam param="Enc_R">[0.01, 0.0, 0.0, 0.01]</rosparam>
    <param name="track_width" value="0.55" type="double"/>
    <param name="tpm_right" value="26500.0" type="double"/>
    <param name="tpm_left" value="27150.0" type="double"/>

    <!-- IMU Initialization -->
    <param name="R_IMU" value="0.000009789186194223082" type="double"/>
  </node>

</launch>
